tools/cloud-build/daily-tests/ansible_playbooks/base-integration-test.yml

```diff
@@ -150,12 +150,19 @@
 
     ## Cleanup and fail gracefully
     rescue:
-    - name: Include rescue from terraform failure
+    - name: Cleanup firewall and infrastructure
       ansible.builtin.include_tasks: "tasks/rescue_terraform_failure.yml"
       vars:
         deployment_name: "{{ deployment_name }}"
         workspace: "{{ workspace }}"
+    - name: Gather logs
```

Comment:
With this ordering we will never get the serial port logs because we have already destroyed the machine. We need to gather logs while they the machine is still up (using the current methodology).

Reply:
Re-shuffling things but ensuring that no step in log gathering is ever allowed to fail (and stop the rescue block).